index.js

```diff
@@ -3983,6 +3983,26 @@
   S.splitOn =
   def('splitOn', {}, [$.String, $.String, $.Array($.String)], splitOn);
 
+  //# splitOnRegex :: RegExp -> String -> Array String
+  //.
+  //. Returns the substrings of its second argument separated by evaluated
+  //. matches of its first argument.
+  //.
+  //. See also [`splitOn`](#splitOn).
+  //.
+  //. ```javascript
+  //. > S.splitOnRegex(/\s+/, 'foo  bar baz   qux')
+  //. ['foo', 'bar', 'baz', 'qux']
+  //. ```
```

Comment:
These are both good suggestions. My reservation about date parsing is that if one is dealing with different separators (`[-/.]`) one is likely to also be dealing with different ordering (YYYYMMDD, DDMMYYYY, MMDDYYYY). My reservation about file paths is that the pattern is a bit noisy (`/[/\\]/`).

How about the following? It's less realistic than your suggested scenarios, but it shows the value of `splitOnRegex` over `splitOn`, and the pattern is quite readable.

```javascript
> S.splitOnRegex(/[,;][ ]*/, 'foo, bar, baz')
['foo', 'bar', 'baz']

> S.splitOnRegex(/[,;][ ]*/, 'foo;bar;baz')
['foo', 'bar', 'baz']
```